### PR TITLE
LSNBLDR-786 Added specific events for page, item and comment actions

### PR DESCRIPTION
**Pull request by adrianfish**
The original ones are also still fired as they are used by SiteStats and
probably other tools. Eventually the old ones should be phased out.

### CODE REVIEW

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java [hunk 1]
```diff
@@ -651,12 +651,15 @@ public boolean saveItem(Object o, List<String>elist, String nowriteerr, boolean
 			if (o instanceof SimplePageItem) {
 				SimplePageItem item = (SimplePageItem)o;
 				EventTrackingService.post(EventTrackingService.newEvent(LessonBuilderEvents.CREATE, "/lessonbuilder/item/" + item.getId(), true));
+				EventTrackingService.post(EventTrackingService.newEvent(LessonBuilderEvents.ITEM_CREATE, "/lessonbuilder/item/" + item.getId(), true));
```
Comment (ottenhoff) [inline lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java]: So we're going to post two events here from the same action? Or are you planning on deleting the previous event?

**Comment by adrianfish** [comment]
Yes, planning on deleting the previous event down the line. I'm not sure what other tools are consuming the current events. I know SiteStats is, but other than that I don't, and I don't want to break anything. So, extra events but less processing time on the event and tools will have less need to parse the reference to actually know what the event is about.

**Comment by bjones86** [comment]
I'm not sure how much benefit this provides, as you still need to parse the reference to get the ID of the item if you want to resolve the object.

Also this has implications for any institutional specific tools which parse and display info on events (which we have here at Western).

**Comment by adrianfish** [comment]
My argument is this: if you want to filter events just based on the event name , you can't as things stand with Lessons. "lessons.create" covers pages, items and comments which to me just adds processing burden, potentially in the event thread itself. My particular case is the bullhorns - I want to setup events for the lessons comments tool and just want to add the particular event to the whitelists. I can't as things stand. @bjones86 Why would having some new events have implications for your specific tools, other than having to add new code to catch them? I've left the current events in and they still get fired, with backwards compat in mind. Lessons has more problems with events than this, too. If you delete a comment you get a "lessons.update" event. If you wanted to find out if that event actually meant delete you'd have to hit the lessons api with the comment id. Deeply inefficient. These new events are just the start :)

**Comment by bjones86** [comment]
Couldn't you write a query that selects items based on the event name and the ref? You could easily filter this way in the DB layer; you don't have to add new event types to accomplish this.

The implications I'm referring to is if this is introduced without widely disseminating/communicating this change, all of a sudden there would be events that custom tools can't resolve.

If others are of the opinion that this is highly valuable, then I stand over-ruled; just providing my two cents here.

**Comment by adrianfish** [comment]
I'm not querying the db - it's an observer. I'm just cherrypicking them on the way past, hence my concerns about event names not being representative. I can just parse the ref in the bullhorns and close this - it just seems really inefficient to have to split strings when you could just have the event right in the first place. Basically, if you add any new tool you can get new events on the bus, new events coming past should not blow up code. I'm just worried about high throughput performance in stuff that cherry picks events and trys not to hog the event thread.

**Comment by adrianfish** [comment]
@ottenhoff So, what do you reckon? Cancel the PR and parse the refererence every time an event comes past? If it's going to bust Brian's stuff I'll just close it.

**Comment by ottenhoff** [comment]
We parse these events also. We are all willing to modify our parsing code if the events are being improved.

I don't like the duplicate events though. I would support this PR if it was a total PR including getting rid of the old events and modifying SiteStats to use the new event names.

**Comment by adrianfish** [comment]
@ottenhoff That is a deal. I don't dig the duplicates either - I just didn't want to break tools watching them.

**Comment by bjones86** [comment]
That sounds good to me; just something for those of us parsing these events to keep in mind when we moving to 12.

**Comment by adrianfish** [comment]
@bjones86 @ottenhoff Search as well ...

**Comment by adrianfish** [comment]
I don't want to merge this until SiteStats is working. I'm getting hibernate errors on a fresh master build, from StatsManagerImpl. I've not been able to test the sitestats bit.

**Pull request merged**